debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,12 +415,19 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.substring(0, gtid.indexOf(":"));
+            if (!gtidDmlSourceFilter.test(uuid)) {
+                ignoreDmlEventByGtidSource = true;
+            }
+        }
```

Comment:
I'm not sure. I assumed every transaction has a GTID event so the value of `ignoreDmlEventByGtidSource` can be appropriately reset there. 

Perhaps the case you are pointing out is possible if there is a GTID transaction followed by a non-GTID transaction, but I'm not sure that's possible... in any case, maybe it's better safe than sorry. I will add that.